src/app/model-actions/model-actions.component.ts

```diff
@@ -0,0 +1,114 @@
+import {Component, OnInit} from '@angular/core';
+import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {ModelField} from '../../interfaces/modelField';
+import {EntityActionsComponent} from '../entity-actions/entity-actions.component';
+import {ActivatedRoute, Params} from '@angular/router';
+import {ProjectService} from '../../services/project-service/project.service';
+import {ToastrService} from 'ngx-toastr';
+import {zip} from 'rxjs';
+import {Model} from '../../interfaces/model';
+import {ModelService} from '../../services/model-service/model.service';
+import {Project} from '../../interfaces/project';
+import {EntitiesEnum} from '../../enums/entities.enum';
+import {FieldTypesEnum} from '../../enums/field-types.enum';
+
+@Component({
+  selector: 'app-model-actions',
+  templateUrl: './model-actions.component.html',
+  styleUrls: ['./model-actions.component.css']
+})
+export class ModelActionsComponent extends EntityActionsComponent implements OnInit {
+  modelForm: FormGroup;
+  public project: Project;
+  public numberFields: number;
+  public options = Object.keys(FieldTypesEnum).filter(key => isNaN(parseInt(key, 10)));
+  public fields: ModelField[] = [{name: 'Some Field', fieldType: 'string'}];
+
+  constructor(private formBuilder: FormBuilder, private activatedRoute: ActivatedRoute,
+              private projectService: ProjectService, private modelService: ModelService,
+              protected toastr: ToastrService) {
+    super(toastr);
+  }
+
+  ngOnInit(): void {
+    const mergeObservables = zip(this.activatedRoute.queryParams, this.projectService.getProjects());
+    mergeObservables.subscribe((params: Params) => {
+      this.project = params[1].find(project => project.name === params[0].projectName);
```

Comment:
your usage of params here is very strange. Why are you accessing number indexes in the params?